### ISSUE TITLE
Include with a variable?

### ISSUE THREAD

**Issue by elado** [jsoverson/preprocess#73]
Is there an option to put a variable in an include?

```
  <!-- @if BLAH -->
    <!-- @include BLAH/file.html -->
  <!-- @endif -->
```

Tried also

```
<!-- @include $BLAH/file.html -->
```

But it doesn't replace the variable, so it doesn't find the file.

**Comment by BendingBender** [comment]
No, not at the moment. As a workaround, you could try to combine this with `@foreach` in the following manner:

```html
<!-- @if BLAH -->
  <!-- @foreach $BLAH in ['somefile'] -->
    <!-- @include $BLAH/file.html -->
  <!-- @endfor -->
<!-- @endif -->
```

But I have no idea whether it will work or not.

**Comment by elado** [comment]
Thanks. I just used `gulp-replace` for now to replace specific variables before preprocess. Does the job.

**Comment by BendingBender** [comment]
Can I close this or do you want it to be implemented?

**Comment by elado** [comment]
I think it'd be a good addition to preprocess, replacing `$VAR` or `${VAR}` with the value before handing `@include`.

**Comment by BendingBender** [comment]
We'll see what we can do :-)

**Comment by karneaud** [comment]
was tis ever implemented?

**Comment by dogancelik** [comment]
Would be a good addition.

Preprocess context:
```js
context: {
  test: 'hello-world'
}
```

File to preprocess:
```js
// @include-static ${test}.js

console.log(1);
```

So that `@include-static ${test}.js` is `@include-static hello-world.js`

**Comment by TheShanMan** [comment]
No ability to use a variable with @include still? I was hoping to generate html files from a common template using this by passing the @include file in via context.
